src/install/resolvers/folder_resolver.zig

```diff
@@ -119,14 +120,13 @@ pub const FolderResolution = union(Tag) {
 
     pub fn readPackageJSONFromDisk(
         manager: *PackageManager,
-        joinedZ: [:0]const u8,
         abs: []const u8,
         version: Dependency.Version,
         comptime features: Features,
         comptime ResolverType: type,
         resolver: ResolverType,
     ) !Lockfile.Package {
-        var package_json: std.fs.File = try std.fs.cwd().openFileZ(joinedZ, .{ .mode = .read_only });
+        var package_json: std.fs.File = try std.fs.cwd().openFile(abs, .{ .mode = .read_only });
```

Comment:
reason why we do the `joinedZ` there was because the system calls all require an extra copy of the filepath to add the extra `\0`

so this says us one memcpy to an extra string buffer

Reply:
Addressed in https://github.com/oven-sh/bun/pull/1764/commits/f1c15797b15c8984626a89ffc211d22221258dfb